template.yaml

```diff
@@ -270,44 +270,49 @@ Resources:
               - sts:TagSession
       Path: /
       Policies:
-        - PolicyName: CuratorAccess
+        - PolicyName: ReadAccess
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - dynamodb:GetItem
+                  - dynamodb:Query
+                  - dynamodb:BatchGetItem
+                  - dynamodb:DescribeTable
+                  - sts:AssumeRole
+                  - sts:TagSession
+                Resource:
+                  - !Join [ "",[ "arn:aws:dynamodb",":",!Ref AWS::Region,":",!Ref AWS::AccountId,":","table/",!Ref DoiRequestsTableName,"/index/",!Ref DoiRequestsIndexName ] ]
+                Condition:
+                  "ForAllValues:StringEquals":
+                    "dynamodb:LeadingKeys":
+                      - "${aws:PrincipalTag/publisherIdentifier}"
+                  StringEquals:
+                    "aws:PrincipalTag/read_doi_request": "READ_DOI_REQUEST"
+
+        - PolicyName: WriteAccess
           PolicyDocument:
             Version: '2012-10-17'
             Statement:
               - Effect: Allow
                 Action:
                   - dynamodb:GetItem
-                  - dynamodb:PutItem
                   - dynamodb:UpdateItem
-                  - dynamodb:Scan
+                  - dynamodb:PutItem
                   - dynamodb:Query
                   - dynamodb:BatchGetItem
                   - dynamodb:DescribeTable
                   - sts:AssumeRole
                   - sts:TagSession
                 Resource:
-                  - !Join
-                    - ""
-                    - - "arn:aws:dynamodb"
-                      - ":"
-                      - !Ref AWS::Region
-                      - ":"
-                      - !Ref AWS::AccountId
-                      - ":"
-                      - "table/"
-                      - !Ref DoiRequestsTableName
-                  - !Join
-                    - ""
-                    - - "arn:aws:dynamodb"
-                      - ":"
-                      - !Ref AWS::Region
-                      - ":"
-                      - !Ref AWS::AccountId
-                      - ":"
-                      - "table/"
-                      - !Ref DoiRequestsTableName
-                      - "/index/"
-                      - !Ref DoiRequestsIndexName
+                  - !Join [ "",[ "arn:aws:dynamodb",":",!Ref AWS::Region,":",!Ref AWS::AccountId,":","table/",!Ref DoiRequestsTableName ] ]
+                Condition:
+                  StringEquals:
+                    "aws:PrincipalTag/approve_doi_request": "APPROVE_DOI_REQUEST"
```

Comment:
Restrict access only to users who have the access rights "APPROVE_DOI_REQUEST" and "REJECT_DOI_REQUEST"